### PR TITLE
Rework Flocktrace creation

### PR DESCRIPTION
**Pull request by FlameArrow57**
[GAME MODES][REWORK]

## About the PR 
This PR reworks Flocktrace costs and adds a max number of Flocktraces you can make.

Flocktraces now are unlocked per a number of total compute reached instead of costing compute.

A Flocktrace limit of floor(Relay compute cost / Flocktrace compute cost) is added.

A Flocktrace counter is added for the ability stats text, so you can see current traces and max traces you can make (doesn't work currently for when possessing drones).

## Why's this needed? 
Allows more room for making buildings, instead of having the conflict of making buildings or Flocktraces.

Prevents making a high number of Flocktraces, which can also be done almost without limit with the Relay converting much of the station when its made.

A trace counter would make it much easier to tell how many traces you have and how many you can make at a glance, and also make that more accessible to Flocktraces.

### CODE REVIEW

**Comment by TobleroneSwordfish** [comment]
Tempted to say we should just scrap traces dying from losing compute along with this as it's very difficult to communicate well to either side and generally just kind of feels bad.

**Comment by FlameArrow57** [comment]
With changes here, having too many traces will shows as ex. Traces: 2/1 in the ability text counter. Also, could always increase the grace period before traces die, and dying traces have overlay on their screen. Could see it making sense either way but idk feels like it goes against Flock scaling in strength with compute. Ideally if Flock is doing fine Flocktraces shouldn't be dying often, main risk points would be near 100 compute increments

**Review by amylizzle — APPROVED** [review]

File: code/datums/flock/flock.dm [hunk 1]
```diff
@@ -18,7 +18,9 @@ var/flock_signal_unleashed = FALSE
 	var/list/priority_tiles = list()
 	var/list/deconstruct_targets = list()
 	var/list/traces = list()
+	var/free_traces = 0
 	var/queued_trace_deaths = 0
+	var/max_trace_count = 0
```
Comment (frawhst) [inline code/datums/flock/flock.dm:23]: Code doc comments pls
don't worry about doing the rest of them, just the ones added here

**Review by frawhst — APPROVED** [review]

File: code/datums/flock/flock.dm (hunk 1 above)
Comment (FlameArrow57) [reply code/datums/flock/flock.dm]: Added if looks okay

**Comment by FlameArrow57** [comment]
Merging from master broke this a bit so will have to test some more

**Comment by FlameArrow57** [comment]
Okay works again

**Review by pali6 — APPROVED** [review]

**Pull request merged**